src/me.rs

```diff
@@ -85,8 +90,12 @@ pub fn motion_estimation(
 
       let mode = PredictionMode::NEWMV;
       let mut tmp_plane = Plane::new(blk_w, blk_h, 0, 0, 0, 0);
-
-      let mut steps = vec![8, 4, 2];
+      let mut steps = vec![];
```

Comment:
We could do something like this:
``` rust
let steps = [8, 4, 2, 1];
let range = if bsize >= BlockSize::BLOCK_32X32 { 0 } else { 1 } .. if fi.allow_high_precision_mv { 4 } else 3;

for step in &steps[range] {
```
It is possibly more compact as long we are trimming a slice :)

Reply:
This is very rustic.
So, the range will be one of four cases:
[8,4,2,1], [4,2,1], [8,4,2], [4,2].
I am now rustic.